src/naive_bayes/bernoulli.rs

```diff
@@ -0,0 +1,275 @@
+use crate::error::Failed;
+use crate::linalg::row_iter;
+use crate::linalg::BaseVector;
+use crate::linalg::Matrix;
+use crate::math::num::RealNumber;
+use crate::math::vector::RealNumberVector;
+use crate::naive_bayes::{BaseNaiveBayes, NBDistribution};
+use crate::preprocessing::binarize;
+
+use serde::{Deserialize, Serialize};
+
+/// Naive Bayes classifier for categorical features
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+struct BernoulliNBDistribution<T: RealNumber> {
+    /// class labels known to the classifier
+    class_labels: Vec<T>,
+    class_priors: Vec<T>,
+    feature_prob: Vec<Vec<T>>,
+}
+
+impl<T: RealNumber, M: Matrix<T>> NBDistribution<T, M> for BernoulliNBDistribution<T> {
+    fn prior(&self, class_index: usize) -> T {
+        if class_index >= self.class_labels.len() {
+            T::zero()
+        } else {
+            self.class_priors[class_index]
+        }
+    }
+
+    fn log_likelihood(&self, class_index: usize, j: &M::RowVector) -> T {
+        if class_index < self.class_labels.len() {
+            let mut likelihood = T::zero();
+            for feature in 0..j.len() {
+                let value = j.get(feature);
+                if value == T::one() {
+                    likelihood += self.feature_prob[class_index][feature].ln();
+                } else {
+                    likelihood += (T::one() - self.feature_prob[class_index][feature]).ln();
+                }
+            }
+            likelihood
+        } else {
+            T::zero()
```

Comment:
I have added this in all the other implementations, but maybe this should be -infinity, right? @VolodymyrOrlov

Reply:
I think we should panic here instead, since `else` branch is unreachable, right?

Reply:
I will remove just this if and else branches since as you say this is unreachable (at least in the public api)